substrate/twin.py

```diff
@@ -0,0 +1,150 @@
+"""Twin module"""
+import hashlib
+from substrateinterface import SubstrateInterface
+from .identity import Identity
+import ipaddress
+
+
+class Twin_Info:
+    """Twin info class"""
+
+    def __init__(self, version: int, id: int, account_id: bytes, ip: ipaddress.IPv4Address, entities: list):
+        self.version = version
+        self.id = id
+        self.account_id = account_id
+        self.ip = ip
+        self.entities = entities
+
+
+class Twin:
+    """Twin class"""
+
+    def __init__(self, substrate: SubstrateInterface, identity: Identity):
+        self.substrate = substrate
+        self.identity = identity
+        self.twin_info = None
+
+    def get(self):
+        """get the twin info for the account ID
+
+        Returns:
+            Twin_Info: the info for a twin
+        """
+        twin_id = get_twin_by_public_key(self.identity.public_key, self.substrate)
+        if twin_id != None:
+            return get_twin_by_id(twin_id, self.substrate)
+
+    def create(self, ip: str):
```

Comment:
for consistency, change the ip in the constructor to be a string and do validations on that string, IPs should be str for ease of use